src/autogluon/bench/eval/scripts/run_generate_clean_openml.py

```diff
@@ -1,69 +1,75 @@
-import argparse
+import logging
+import os
 
 import pandas as pd
+import typer
 
 from autogluon.bench.eval.evaluation.constants import FRAMEWORK
 from autogluon.bench.eval.evaluation.preprocess import preprocess_openml
 from autogluon.common.savers import save_pd
 
+app = typer.Typer()
+logger = logging.getLogger(__name__)
 
-def clean_and_save_results(
-    run_name,
-    results_dir="data/results/",
-    results_dir_input=None,
-    results_dir_output=None,
-    file_prefix="results_automlbenchmark",
-    run_name_in_input_path=True,
-    constraints=None,
-    out_path_prefix="openml_ag_",
-    out_path_suffix="",
-    framework_suffix_column="constraint",
+
+@app.command()
+def clean_amlb_results(
+    benchmark_name: str = typer.Option(
+        ..., help="Benchmark name populated by benchmark run, in format <benchmark_name>_<timestamp>"
+    ),
+    results_dir: str = typer.Option("data/results/", help="Root directory of raw and prepared results."),
+    results_input_dir: str = typer.Option(
+        None,
+        help="Directory of the results file '<file_prefix><constraint_str><benchmark_name_str>.csv' getting cleaned. Can be an S3 URI. If not provided, it defaults to '<results_dir>input/raw/'",
+    ),
+    results_output_dir: str = typer.Option(
+        None,
+        help="Output directory of cleaned file. Can be an S3 URI. If not provided, it defaults to '<results_dir>input/prepared/openml/'",
+    ),
+    file_prefix: str = typer.Option("results_automlbenchmark", help="File prefix of the input results files."),
+    benchmark_name_in_input_path: bool = False,
+    constraints: str = typer.Option(
+        None,
+        help="List of AMLB constraints, refer to https://github.com/openml/automlbenchmark/blob/master/resources/constraints.yaml",
+    ),
+    out_path_prefix: str = typer.Option("openml_ag_", help="Prefix of result file."),
+    out_path_suffix: str = typer.Option("", help="suffix of result file."),
```

Comment:
Can we still call this as a function in python code after this change? If not, that might be problematic.

Reply:
Yes we can, though the only tricky thing is that we need to supply "None" or default values for all in the function arguments. I'm thinking about another solution, which is to write a `typer` function wrapper around these functions. That might solve the problem of providing both CLI interface and normal function usage.

Reply:
I would like to preserve normal function usage if at all possible, otherwise it will become awkward to chain these function calls in higher level code.

Reply:
Alternatively, if there is a design pattern that suggests something different from what I am describing, I'd be interested to learn more.